clouddrift/wavelet.py

```diff
@@ -0,0 +1,518 @@
+    # enforce length 1 for first axis if order=1 (no squeezing)
+    wave = np.moveaxis(wave, [0, 1, 2], [2, 0, 1])
+    wavefft = np.moveaxis(wavefft, [0, 1, 2], [2, 0, 1])
+    return wave, wavefft
+
+
+def _morsewave_first_family(
+    fact: float,
+    n: int,
+    ga: float,
+    be: float,
+    norm_rad_freq: np.ndarray,
+    wavezero: np.ndarray,
+    order: Optional[int] = 1,
+    norm: Optional[str] = "bandpass",
+) -> np.ndarray:
+    """
+    Derive first family wavelet. Internal use only.
+    """
+    r = (2 * be + 1) / ga
+    c = r - 1
+    L = np.zeros_like(norm_rad_freq, dtype=np.float64)
+    wavefft1 = np.zeros((np.shape(wavezero)[0], order))
+
+    for i in np.arange(0, order):
+        if norm == "energy":
+            A = morseafun(ga, be, order=i + 1, norm=norm)
+            coeff = np.sqrt(1 / fact) * A
+        elif norm == "bandpass":
+            if be != 0:
+                coeff = np.sqrt(np.exp(_lgamma(r) + _lgamma(i + 1) - _lgamma(i + r)))
+            else:
+                coeff = 1
+
+        index = slice(0, int(np.round(n / 2)))  # how to define indices?
+        L[index] = _laguerre(2 * norm_rad_freq[index] ** ga, i, c)
+        wavefft1[:, i] = coeff * wavezero * L
+
+    return wavefft1
+
+
+def morsefreq(
+    ga: Union[np.ndarray, float],
+    be: Union[np.ndarray, float],
+) -> Union[Tuple[np.ndarray], Tuple[float]]:
+    """
+    Frequency measures for generalized Morse wavelets. This functions calculates
+    three different measures of the frequency of the lowest-order generalized Morse
+    wavelet specified by parameters gamma (ga) and beta (beta).
+
+    Note that all frequency quantities here are *radian* as in cos(omega t)and not
+    cyclic as in np.cos(2 np.pi f t).
+
+    For be=0, the "wavelet" becomes an analytic lowpass filter, and fm
+    is not defined in the usual way.  Instead, fm is defined as the point
+    at which the filter has decayed to one-half of its peak power.
+
+    For details see Lilly and Olhede (2009).  Higher-order properties of analytic
+    wavelets.  IEEE Trans. Sig. Proc., 57 (1), 146--160.
+
+    Parameters
+    ----------
+    ga: np.ndarray or float
+       Gamma parameter of the wavelet.
+    be: np.ndarray or float
+       Beta parameter of the wavelet.
+
+    Returns
+    -------
+    fm: np.ndarray
+        The modal or peak frequency.
+    fe: np.ndarray
+        The "energy" frequency.
+    fi: np.ndarray
+        The instantaneous frequency at the wavelet center.
+
+    Examples
+    --------
+    To write.
+
+    See Also
+    --------
+    :func:`morsewave`
+    """
+    # add test for type and shape in case of ndarray
+    fm = np.where(
+        be == 0, np.log(2) ** (1 / ga), np.exp((1 / ga) * (np.log(be) - np.log(ga)))
+    )
+
+    fe = 1 / (2 ** (1 / ga)) * _gamma((2 * be + 2) / ga) / _gamma((2 * be + 1) / ga)
+
+    fi = _gamma((be + 2) / ga) / _gamma((be + 1) / ga)
+
+    return fm, fe, fi
+
+
+def morsespace(
```

Comment:
This name is good but may want a _

```suggestion
def morse_space(
```